botwo/botwo.py

```diff
@@ -0,0 +1,200 @@
+import fnmatch
+import botocore
+
+COPY_METHODS = {"copy", "copy_object", "copy_upload_part"}
+LIST_METHODS = {"list_objects", "list_objects_v2", "list_object_version"}
+
+
+class PaginatorWrapper(object):
+    """ This class wraps boto paginator """
+
+    def __init__(self, mapping, config, kwargs):
+        self.mapping = mapping
+        self.config = config
+        self.paginators = dict()
+        for client in self.mapping:
+            self.paginators[client] = self.mapping[client].get_paginator(kwargs["operation_name"])
+        self.default = self.paginators.get("default")
+
+    def paginate(self, **kwargs):
+        for profile in self.config:
+            mapping = self.config[profile]
+
+            if fnmatch.fnmatch(kwargs["Bucket"], mapping.get("source_bucket_pattern")):
+                if kwargs["Prefix"]:
+                    if mapping.get("source_key_pattern"):
+                        if not fnmatch.fnmatch(kwargs["Prefix"], mapping.get("source_key_pattern")):
+                            continue
+                    if mapping.get("mapped_prefix"):
+                        kwargs["Prefix"] = mapping.get("mapped_prefix") + kwargs["Prefix"]
```

Comment:
AFAIU Python will give a `KeyError` here unless `kwargs` has a `Prefix` key.  So this only checks whether `Prefix` is empty, right?

So I think this is a bug: if `Prefix` is empty then it does not prepend `mapped_prefix`, and the prefix stays empty.  E.g. (I think) listing the empty prefix will be wrong if the default has a non-empty `mapped_prefix`.

I think this simpler code is better because it fixes the bug that I think we have:
```suggestion
                if mapping.get("source_key_pattern"):
                    if not fnmatch.fnmatch(kwargs["Prefix"], mapping.get("source_key_pattern")):
                        continue
                if mapping.get("mapped_prefix"):
                    kwargs["Prefix"] = mapping.get("mapped_prefix") + kwargs["Prefix"]
```